libraries/AP_WayBack/AP_WayBack.cpp

```diff
@@ -0,0 +1,818 @@
+/*
+
+this library collects points of drone position, records them in memory and eliminates 
+unnecessary data - redundant points and loops, so it always knows 
+the shortest way to home via visited points.
+ 
+ Copyright night_ghost@ykoctpa.ru 2017
+ 
+ license GPLV3 or above
+   
+*/
+
+#include "AP_WayBack.h"
+
+extern const AP_HAL::HAL& hal;
+
+AP_WayBack::Point* AP_WayBack::points;
+uint16_t          AP_WayBack::max_num_points=0; // size of storage
```

Comment:
fully-static it can't be because it should work in Ardupilot API. I can add a check to constructor, but this will be the same waste of flash: we have just one AHRS so if anyone (who???) creates 2nd instances then both points will be from one source and will be effectively eliminated. 
Moreover this library eats out a lot of memory for points buffer so to allow this memory to be per-instance is a very bad idea.
Codebase is just an existing solutions of some tasks, and not the fact that the ways to solve them will suit all tasks.